src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Commands/ObjectBrowserCommand.cs

```diff
@@ -0,0 +1,142 @@
+
+using Microsoft.VisualStudio.ProjectSystem.Input;
+using Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies;
+using Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections;
+using Microsoft.VisualStudio.Shell.Interop;
+
+using Flags = Microsoft.VisualStudio.ProjectSystem.Tree.Dependencies.DependencyTreeFlags;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.Commands;
+
+/// <summary>
+/// Implements Visual Studio's "Open in Object Browser" command for dependencies tree nodes.
+/// </summary>
+/// <remarks>
+/// Items that can be opened in the Object Browser must advertise the fact.
+/// The approach depends on the type of node:
+/// <list type="bullet">
+/// <item>
+/// Direct project dependencies, modelled as <see cref="IDependency"/>, must also implement <see cref="IDependencyWithBrowseObject"/>,
+/// include the <see cref="Flags.SupportsObjectBrowser"/> tree flag, and populate the <c>BrowsePath</c> property on the browse object.
+/// </item>
+/// <item>
+/// Relatable items (attached sub-nodes) modelled as <see cref="IRelatableItem"/>, must implement <see cref="IObjectBrowserItem"/>.
+/// </item>
+/// </list>
+/// </remarks>
+[ProjectCommand(VSConstants.CMDSETID.StandardCommandSet2K_string, (long)VSConstants.VSStd2KCmdID.VIEWREFINOBJECTBROWSER)]
+[AppliesTo(ProjectCapability.DependenciesTree)]
+[Order(Order.Default)]
+[method: ImportingConstructor]
+internal sealed class ObjectBrowserCommand(UnconfiguredProject project, IVsUIService<SVsObjBrowser, IVsNavigationTool> navigationTool) : AbstractSingleNodeProjectCommand
+{
+    // This code needs a little explanation. Consider a tree like this:
+    //
+    // - Dependencies
+    //   - Packages
+    //     - MyPackage
+    //       - MyPackage.dll
+    // - Frameworks
+    //   - Microsoft.NETCore.App
+    //     - Microsoft.CSharp
+    //
+    // The "MyPackage" and "Microsoft.NETCore.App" nodes are top-level dependencies, and they have IProjectTree nodes.
+    //
+    // The "MyPackage.dll" and "Microsoft.CSharp" nodes are transitive nodes, added via the "AttachedCollections" API
+    // (IRelatableItem) and do not have IProjectTree nodes.
+    //
+    // When the user right-clicks on an IRelatableItem node, CPS is called with the IProjectTree node of the closest
+    // ancestor, which is not the node the user clicked on. There's no easy way to thread that state through on the
+    // call stack, so we use a static collection on the menu controller that's only populated while the context menu
+    // is open. If it has items here, we know we're in the transitive case and will attempt to handle the command.
+
+    protected override async Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string? commandText, CommandStatus progressiveStatus)
+    {
+        string? path = await TryGetAssemblyPathAsync(node);
+        
+        if (path is not null)
+        {
+            // We handle this .
```

Comment:
nit: extra whitespace here.

Reply:
Oh yeah, looks like I never finished writing this :)

Reply:
```suggestion
            // We handle this.
```